### PR TITLE
(feat-rocksdb) Working set prep - refactor RocksDbStore into DbStore and RocksDbTrieStore

### PR DESCRIPTION
**Pull request by dwerner**
This PR is intended to refactor some database access code to be more general and reusable. Other work on working-set will be based on this.

### CODE REVIEW

File: execution_engine/src/storage/db_store.rs [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+    }
+
+    /// Working set column family.
+    pub fn working_set_column_family(&self) -> Result<Arc<BoundColumnFamily<'_>>, error::Error> {
+        self.db.cf_handle(WORKING_SET_COLUMN_FAMILY).ok_or_else(|| {
+            error::Error::UnableToOpenColumnFamily(WORKING_SET_COLUMN_FAMILY.to_string())
+        })
+    }
+
+    /// Trie store iterator.
```
Comment (goral09) [inline execution_engine/src/storage/db_store.rs]: Can you include in the comment what it is iterating over? It's not iterating over elements in the trie store - trie nodes, is it?

File: execution_engine/src/storage/global_state/db.rs [hunk 2]
```diff
@@ -267,7 +266,7 @@ impl DbGlobalState {
 
 fn memoized_find_missing_descendants(
     value_bytes: Bytes,
-    rocksdb: RocksDbStore,
+    rocksdb_store: &RocksDbTrieStore,
```
Comment (goral09) [inline execution_engine/src/storage/global_state/db.rs]: I don't think we have to include the type of the DB in the name - maybe `trie_store` is enough.

File: execution_engine/src/storage/global_state/db.rs [hunk 3]
```diff
@@ -99,12 +99,12 @@ pub struct DbGlobalStateView {
     /// Root hash of this "view".
     root_hash: Digest,
     /// Handle to rocksdb.
-    rocksdb_store: RocksDbStore,
+    rocksdb_store: RocksDbTrieStore,
```
Comment (goral09) [inline execution_engine/src/storage/global_state/db.rs]: This `DbGlobalStateView` looks like a good candidate for a working set?

File: execution_engine/src/storage/trie_store/db.rs [hunk 4]
```diff
@@ -142,147 +120,103 @@ impl Store<Digest, Trie<Key, StoredValue>> for ScratchTrieStore {
...
-
-        // recommended to match # of cores on host.
-        db_opts.increase_parallelism(num_cpus::get() as i32);
-
-        db_opts.create_missing_column_families(true);
-        db_opts.create_if_missing(true);
-
-        db_opts
+    /// Access the underlying DbStore.
+    pub fn get_db_store(&self) -> &DbStore {
```
Comment (goral09) [inline execution_engine/src/storage/trie_store/db.rs:209]: Should we be exposing the underlying RocksDB for consumption in the public API? Can we hide it behind domain-specific methods?

**Review by goral09 — APPROVED** [review]

File: execution_engine/src/storage/global_state/db.rs (hunk 3 above)
Comment (dwerner) [reply execution_engine/src/storage/global_state/db.rs]: The `...GlobalStateView` stuff is code consumed by the EE. I agree we need access to the working set during execution.

File: execution_engine/src/storage/trie_store/db.rs (hunk 4 above)
Comment (dwerner) [reply execution_engine/src/storage/trie_store/db.rs:209]: We can make this `pub(crate)` for now. Done in [6c20f4a](https://github.com/casper-network/casper-node/pull/2999/commits/6c20f4a0f076a2a375dc7d0f332871c085825545)

File: execution_engine/src/storage/global_state/db.rs (hunk 2 above)
Comment (dwerner) [reply execution_engine/src/storage/global_state/db.rs]: Done in [6c20f4a](https://github.com/casper-network/casper-node/pull/2999/commits/6c20f4a0f076a2a375dc7d0f332871c085825545)

File: execution_engine/src/storage/db_store.rs (hunk 1 above)
Comment (dwerner) [reply execution_engine/src/storage/db_store.rs]: Yes, it is, however I intended to move this to `RocksDbTrieStore`. Done in [6c20f4a](https://github.com/casper-network/casper-node/pull/2999/commits/6c20f4a0f076a2a375dc7d0f332871c085825545)

**Pull request merged**